discovery-infra/tests/test_logs.py

```diff
@@ -10,112 +10,78 @@
+        new_cluster.prepare_for_install(nodes=nodes)
         started_cluster_install_at = time.time()
-        self.start_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        new_cluster.start_install()
         # Cancel cluster install once at least one host has been rebooted
-        self.wait_for_one_host_to_boot_during_install(cluster_id=cluster_id, api_client=api_client)
-        self.cancel_cluster_install(cluster_id=cluster_id, api_client=api_client)
-        assert self.is_cluster_in_cancelled_status(
-            cluster_id=cluster_id,
-            api_client=api_client
-        )
+        new_cluster.wait_for_at_least_one_host_to_boot_during_install()
+        new_cluster.cancel_install()
+        assert new_cluster.is_in_cancelled_status()
         # download logs into a file. At this point logs exist and we expect them to
         # be from the hosts that finished reboot without controller logs
         '''verify logs are uploaded during the first installation'''
         path = "/tmp/test_on_restart_logs.tar"
         expected_min_log_num = 1
-        self.verify_logs_uploaded(cluster_id, api_client, path, expected_min_log_num)
-
+        self._verify_logs_uploaded(new_cluster.id, api_client, path, expected_min_log_num)
         # reset cluster and that boot the hosts again from ISO and wait for re-discovery
-        '''reset cluster and reboot hosts'''
-        self.reset_cluster_and_wait(cluster(), api_client, nodes)
+        self._reset_cluster_and_wait(new_cluster, nodes)
         # re-install the cluster
-        '''second installation'''
         started_cluster_install_at = time.time()
-        self.install_cluster_and_wait(cluster_id, api_client)
-
+        self._install_cluster_and_wait(new_cluster)
         # verify that the logs are current after re-install and were not left from
         # the previous attempt
         '''verify second upload of logs'''
-        logs_collected_at = utils.get_logs_collected_at(api_client, cluster_id)
+        logs_collected_at = utils.get_logs_collected_at(api_client, new_cluster.id)
         expected_min_log_num = len(nodes) + 1
-        self.verify_logs_uploaded(cluster_id, api_client, path, expected_min_log_num)
-        self.verify_logs_are_current(started_cluster_install_at, logs_collected_at)
+        self._verify_logs_uploaded(new_cluster.id, api_client, path, expected_min_log_num)
+        self._verify_logs_are_current(started_cluster_install_at, logs_collected_at)
 
-    def verify_no_logs_uploaded(self, cluster_id, api_client, path):
+    def _verify_no_logs_uploaded(self, cluster_id, api_client, path):
         with pytest.raises(ApiException) as ex:
             api_client.download_cluster_logs(cluster_id, path)
         assert "No log files" in str(ex.value)
 
-    def verify_logs_uploaded(self, cluster_id, api_client, path, expected_min_log_num):
+    def _verify_logs_uploaded(self, cluster_id, api_client, path, expected_min_log_num):
```

Comment:
add download_logs to Cluster and use that, it will probably be useful  for other tests